rust/sbp/src/sbp_tools.rs

```diff
@@ -7,7 +9,47 @@ use crate::{
     time::{GpsTimeError, MessageTime, RoverTime},
 };
 
-use crate::messages::SBP;
+use crate::messages::{MessageType, SBP};
+
+pub struct Dispatcher {
+    callbacks: HashMap<u16, Vec<Box<dyn FnMut(SBP)>>>,
```

Comment:
Probably an over optimization (and would need some benchmarking to prove/disprove), but I think a `Vec<Box<dyn FnMut(SBP)>>` might be faster, you'd also need to make the the closure swallow the error on a failed conversion. Maybe doing all those conversions is too expensive?